app/code/Magento/InventorySales/Plugin/CatalogInventory/StockManagement/ProcessBackItemQtyPlugin.php

```diff
@@ -86,13 +87,22 @@ public function __construct(
      *
      * @SuppressWarnings(PHPMD.UnusedFormalParameter)
      */
-    public function aroundBackItemQty(StockManagement $subject, callable $proceed, $productId, $qty, $scopeId = null)
-    {
+    public function aroundBackItemQty(
+        StockManagement $subject,
+        callable $proceed,
+        $productId,
+        $qty,
+        $scopeId = null
+    ): bool {
         if (null === $scopeId) {
             throw new LocalizedException(__('$scopeId is required'));
         }
 
-        $productSku = $this->getSkusByProductIds->execute([$productId])[$productId];
+        try {
+            $productSku = $this->getSkusByProductIds->execute([$productId])[$productId];
+        } catch (InputException $e) {
```

Comment:
could you please provide a description why we consider swallowing of exception as an appropriate way of handling

Btw, why do we throw InputException out of 
```
        if (!empty($notFoundedIds)) {
            throw new InputException(
                __('Following products with requested ids were not found: %1', implode($notFoundedIds, ', '))
            );
        }
```
but not `\Magento\Framework\Exception\NoSuchEntityException`

Reply:
As we decided that the inventory doesn't depend on the catalog (these two system are independent) it is necessary for us to have an ability process orders even with deleted or non-existing products